distro/wildfly/subsystem/src/main/java/org/camunda/bpm/container/impl/jboss/service/ProcessApplicationStopService.java

```diff
@@ -104,6 +104,9 @@ public void stop(StopContext arg0) {
       if(reference != null) {
         reference.release();
       }
+      if (processApplication != null ) {
+        processApplication = null;
+      }
```

Comment:
❓ Can we somehow access the `ProcessApplicationReferenceImpl` here to clear the weak reference?

Reply:
Will work on that.

Reply:
Pushed adjustment in 70c5a20.

Reply:
I debugged it a little more and was thinking about whether we can keep everything as before (revert this fix) but get the `ProcessApplicationInterface` from the supplier in the `ProcessApplicationStopService#start` method and store it in a member variable. At this point in time, the context destroyed hasn't been fired, so it's still possible to get the object. We would then have access to it on `ProcessApplicationStopService#stop`. Is it a problem that we have a strong reference in `ProcessApplicationStopService` on it? My first guess would be no since a `ProcessApplicationStopService` instance should exist for each deployment and is GC on stop, isn't it?

Reply:
I will try it out.
To reviewer (Tobi): even that I am still working on these last bits, I still hope to receive feedback on the rest of the changes.

Reply:
I will add a link to this discussion in the ticket with short summary if we stumble upon the issue on next update.
Since we revert the `#addDependency` changes, the workaround will not be needed anymore. Therefore I am closing the discussion.